File: Day_8_Dictionaries_and_Maps/Solution.swift
// Enter your code here. Read input from STDIN. Print output to STDOUT
var phoneBook = [String: String]()
var n = Int(readLine()!) // assume you enter your Name

for _ in 1 ... n! {
    let arg = readLine()!.split(separator: " ")
    phoneBook[String(arg[0])] = String(arg[1])
}

// var search = String(readLine()!)
// while search != "" {
//     // for _ in 1 ... n! {
//     if let _ = myMap[search] {
//         // now val is not nil and the Optional has been unwrapped, so use it
//         print(search + "=" + myMap[search]!)
//     } else {
//         print("Not found")
//     }
//     search = String(readLine()!)
// }

if let input = readLine() {
        var name = String(input)

        repeat {
            if let phoneNumber = phoneBook[name] {
                print("\(name)=\(phoneNumber)")
            } else {
                print("Not found")
            }

            if let input = readLine() {
                name = String(input)
            } else {
                name = ""
            }

        } while name.count > 0
    }

File: Day_8_Dictionaries_and_Maps/C++/Solution.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
using namespace std;

int main()
{
  /* Enter your code here. Read input from STDIN. Print output to STDOUT */
  int n;
  cin >> n;
  // empty map container
  map<string, int> phoneBook;
  vector<string> searches;

  for (int i = 0; i < n; i++)
  {
    // insert elements in random order
    string name;
    cin >> name;

    int number;
    cin >> number;
    phoneBook.insert(pair<string, int>(name, number));
  }
  string search;
  while (cin >> search)
  {
    // searches.push_back(search);
    if (phoneBook.find(search) == phoneBook.end())
    {
      cout << "Not found" << endl;
    }
    else
    {
      cout << phoneBook.find(search)->first << "=" << phoneBook.find(search)->second << endl;
    }
  }
  return 0;
}
